engine/table/src/main/java/io/deephaven/engine/table/impl/util/DynamicTableWriter.java

```diff
@@ -527,7 +594,12 @@ void writeToColumnSource() {
 
         @Override
         public void set(Object value) {
-            setInt(value == null ? QueryConstants.NULL_INT : (Integer) value);
+            setInt(value == null ? QueryConstants.NULL_INT : (int) value);
```

Comment:
Does casting as a primitive `int` make us more permissive here than before? I'm not sure if there might be some boxed -> primitive logic that is different than casting to explicit box.

I could check myself, but what happens in the following cases:

```java
int x1 = (int)Long.valueOf(1);
int x2 = (Integer)Long.valueOf(1);
int x3 = (int)Short.valueOf(1);
int x4 = (Integer)Short.valueOf(1);
```

I'm *assuming* that `x2` and `x4` should fail - does `x1` and `x3` fail similarly?

Ditto for all other `set` changes.